src/lib/sidenav/sidenav.ts

```diff
@@ -137,15 +137,24 @@ export class MdSidenav implements AfterContentInit, OnDestroy {
       }
     });
 
-    this.onClose.subscribe(() => {
+    this.onClose.subscribe(() => this._restoreFocus());
+  }
+
+  /**
+   * If focus is currently inside the sidenav, restores it to where it was before the sidenav
+   * opened.
+   */
+  private _restoreFocus() {
+    let activeEl = document.activeElement;
```

Comment:
It seems like we should be injecting the `Document` rather than using the global (like we do elsewhere). Is there a reason why we aren't doing it here?